src/orient/rational-calc.orient

```diff
@@ -0,0 +1,550 @@
+    describe(replica_inclusions, "Constraints to check all the inclusions required in a window in CommR")
+    describe(window_replica_inclusions,  "Constraints to check all the inclusions required for all windows in CommR")
+    describe(window_inclusion, "Constraints to check a single inclusion proof in a window")
+    describe(window_inclusions, "Constraints to check all the required inclusion proofs in a window")
+    describe(column_leaf, "Constraints to check a single leaf generation from a column in a window")
+    describe(column_leaves, "Constraints to check all the leaves generation required in a window")
+    describe(window_column_leaves, "Constraints to check all the leaves generation required for all windows")
+
+    // D
+    // Data layer committed with merkle tree
+    data_inclusion = merkle_tree_datahash_constraints * log2(nodes)
+    data_inclusions = data_inclusion * porep_challenges
+    window_data_inclusions = data_inclusion * porep_challenges * windows
+
+    // C
+    window_inclusion = merkle_tree_hash_constraints * log2(window_nodes)
+    window_inclusions =  window_inclusion * porep_challenges * (graph_parents + 1)
+    column_leaf = column_leaf_hash_constraints * stacked_layers
+    column_leaves = column_leaf * porep_challenges * (graph_parents + 1)
+    window_column_leaves = column_leaves * windows
+    labeling_proof = kdf_constraints * graph_parents
+    labeling_proofs = labeling_proof * stacked_layers * porep_challenges
+    window_labeling_proof = labeling_proofs * windows
+
+    // Q and R
+    replica_inclusion = merkle_tree_hash_constraints * commr_tree_depth 
+    replica_inclusions = replica_inclusion * porep_challenges
+    window_replica_inclusions = replica_inclusions * windows
+
+    porep_snark_partitions = porep_snark_constraints / porep_snark_partition_constraints
+    porep_snark_size = porep_snark_partitions * snark_size
+    porep_snark_size_kib = porep_snark_size / b_in_kib
+    porep_snark_time = snark_constraint_time * porep_snark_constraints
+
+    Constraints (WindowEncoding):
+      porep_snark_constraints = window_inclusions + window_column_leaves + window_labeling_proof + window_data_inclusions + window_replica_inclusions*2
+
+    Constraints (SectorEncoding):
+      porep_snark_constraints = window_inclusions + window_column_leaves + window_labeling_proof + data_inclusions + replica_inclusions*2
+
+    Constraints (Wrapping):
+      porep_snark_constraints = window_inclusions + window_column_leaves + window_labeling_proof + data_inclusions + replica_inclusions*2
+
+    // Column comitting data or replica. 
+
+ProofOfReplication (StackedChungParameters):
+  describe(chung_a, "Chung A")
+  describe(chung_c, "Chung C")
+  describe(chung_delta, "Chung delta")
+  describe(chung_sigma, "Chung sigma")
+
+  chung_a = spacegap - (chung_c / (chung_c -1)) * chung_delta + ((chung_c-1)/ chung_c) * (chung_sigma - 1)
+
+  stacked_layers = (log2(regeneration_fraction / chung_a) / log2(chung_c)) + 1
```

Comment:
This is not conservative at all : we want to set 2 or 3 times this number to be on the safe side, such that even if the cost of encoding goes a bit, it's still safe. We might want to have an extra factor like `marginFactor` that we multiply with regeneration_fraction here.